src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -590,17 +648,81 @@ Segment::write_ertr::future<> ZNSSegment::write(
   seastore_off_t offset, ceph::bufferlist bl)
 {
   LOG_PREFIX(ZNSSegment::write);
-  if (offset < write_pointer || offset % manager.metadata.block_size != 0) {
-    ERROR("invalid segment write on segment {} to offset {}",
+  if (offset != write_pointer || offset % manager.metadata.block_size != 0) {
+    ERROR("Invalid write on segment {} to offset {}",
       id,
       offset);
     return crimson::ct_error::invarg::make();
   }
-  if (offset + bl.length() > manager.metadata.segment_size)
+  if (offset + bl.length() > manager.metadata.segment_capacity)
     return crimson::ct_error::enospc::make();
   
   write_pointer = offset + bl.length();
   return manager.segment_write(paddr_t::make_seg_paddr(id, offset), bl);
 }
 
+Segment::write_ertr::future<> ZNSSegment::write_padding_bytes(
+  seastore_off_t offset, size_t padd_bytes, seastore_off_t tail_offset)
+{
+  LOG_PREFIX(ZNSSegment::write_padding_bytes);
+  DEBUG("Writing {} padding bytes to segment {} at wp {}",
+                                                padd_bytes, id, write_pointer);
+
+  if (padd_bytes == 0)
```

Comment:
Please wrap the if block with {}:
```c++
if (padd_bytes == 0) {
  return write_ertr::now();
}
```

Same to the other places.

Reply:
Ok, changed in other places, where I am touching.